modules/c++/math.linear/include/math/linear/Matrix2D.h

```diff
@@ -1329,9 +1329,35 @@ template<typename _T> inline
     if (mx.rows() == 3)
         return inverse3x3<_T>(mx);
         // TODO Add 4x4
-    
+
     return inverseLU<_T>(mx);
 }
+
+/*!
+ * Calculate the left inverse
+ * \param mx The matrix to find the inverse of. Mx is assumed to
+ * have full row rank
```

Comment:
Can you spell out here and in the rightInverse() method what "full row rank" / "full column rank" means (i.e. more cols than rows, etc.)?